### ISSUE TITLE
Set up a way to replicate content to the internal server

### ISSUE THREAD

**Issue by limikael** [tunapanda/TI-wp-content-theme#6]
Do research to find a good way to replicate pages and H5P content from one server to another. It should be something that runs automatically in the background.

**Comment by limikael** [comment]
It would be possible with bash scripts and cron jobs and such. It would be better, however, to create it with standardized wordpress components. Investigate if such plugins exist. If there is nothing like that, how much work would it be to create what we need? If it can't be created, then as a last resort we will have to rely on bash scripts.

For reference, there is https://www.drupal.org/project/replicate for Drupal.

**Comment by c3c1l1a** [comment]
I have done a bit of research (Don't know if I was exhaustive enough) and this is what I came up with. I have found a few plugins that are almost doing what we want 
1) Dublicator --> https://wordpress.org/plugins/duplicator/ 
This is pretty efficient in creating a duplicate of a site to a new one. I don't know if it deals with keeping the two wp instances in sync. This is not exactly what we want.
2) WP Migrate DB Pro --> https://wordpress.org/plugins/wp-migrate-db/
This is good for migrating a db from one wp instance to another. This plugin is just limited to migrating the database. The process is not automatic for subsequent db synchronization. 
3) wp crontrol --> https://wordpress.org/plugins/wp-crontrol/installation/
This is just lets you view and control what's happening in the WP-Cron system. 

This is what I have in mind. Not sure about the whole thing though. 
All the three plugins are doing bits and pieces of what I think we want. Why not combine the ideas of the  three and build a feature in our theme that can implement that. A plugin would be another option.

But before this is put into consideration may be someone may have done better research and found exactly what we want.

**Comment by limikael** [comment]
Ah ok... Would be cool to create a plugin for this actually... :) And if nothing exists I guess there wouldn't be any other option... We can ask around a bit first though, maybe @Jakeii knows something?

**Comment by c3c1l1a** [comment]
Ok sounds like a good idea.

**Comment by limikael** [comment]
For reference... Here is something that also does a similar thing...

https://wordpress.org/plugins/pushlive/

But it seems it only works for two sites that are stored on the same server...

**Comment by Jakeii** [comment]
using a cron, sftp and rsync sounds pretty good to me? avoid downloading the whole thing over and over..

But by doing a db/file level duplication, you would loose any changes on the local server, would a 'post level' import be better?

**Comment by limikael** [comment]
Yeah I think "post level" would be the best... Thinking how to create a kind of distributed version control system actually, because that would be interesting... So for one wordpress site you can set a "remote" (like you do in git) and then pull and push changes... If nothing like that exists it could be worthwhile...

Found this which could be a component: https://github.com/luqui/merge3

**Comment by limikael** [comment]
Work started here: https://github.com/tunapanda/wp-remote-sync/

**Comment by limikael** [comment]
Have an RC now of https://github.com/tunapanda/wp-remote-sync/

Waiting for Wordpress to approve putting in in the Wordpress plugin repo.

**Comment by limikael** [comment]
Waiting for the key authentication to be implemented before deploying...

**Comment by limikael** [comment]
Alright!

It's now up and running...

http://192.168.0.254/learning.tunapanda.org 

from inside Tunapanda, is synced with:

http://learning.tunapanda.org

**Comment by munga4** [comment]
Problem found while testing.
https://github.com/tunapanda/wp-remote-sync/issues/14

**Comment by limikael** [comment]
Oh damn... Currently there is a problem:

```
Status: post
** Error **
no revision in RemoteResource constr...
#0 /home/micke/repo/wp-remote-sync/src/model/RemoteResource.php(127): RemoteResource->__construct('post', 'kcse-past-paper...', NULL)
#1 /home/micke/repo/wp-remote-sync/src/model/SyncResource.php(317): RemoteResource::fetchAllForType('post')
#2 /home/micke/repo/wp-remote-sync/src/controller/RemoteSyncOperations.php(75): SyncResource::findAllForType('post', 3)
#3 [internal function]: RemoteSyncOperations->status()
#4 /home/micke/repo/wp-remote-sync/src/controller/RemoteSyncOperations.php(42): call_user_func(Array)
#5 /home/micke/repo/wp-remote-sync/op.php(16): RemoteSyncOperations->handleOperation('Status')
#6 {main}
```

Looking in to it...

**Comment by limikael** [comment]
Ok fixed, in version 0.0.20...

The important thing to test when verifying this issue is that http://192.168.0.254/learning.tunapanda.org/ can sync with http://learning.tunapanda.org

So I think the main test we should perform is:
- Check that currently, http://192.168.0.254/learning.tunapanda.org/ is in sync. Test a few random swagpaths here and there, 5 or so...
- Make a change to one swagpath on http://learning.tunapanda.org, e.g. in something under the "Test track": http://learning.tunapanda.org/courses/testing/
- Do "status" on http://192.168.0.254/learning.tunapanda.org/ to see that the change is detected.
- Do "pull" on http://192.168.0.254/learning.tunapanda.org/. This should cause the change to be downloaded and applied locally.

If this works I think we can see the "definition of done" as having been reached for this issue, and we can close it.

**Comment by mrlarso** [comment]
From 192.168.0.254/learning.tunapanda.org:

**Pull** reports 'done' but not all content on local instance matches learning.tunapanda.org

**Push** repeatedly freezes  (15+ minutes as below with no change) and changes not reflected on learning.tunapanda.org

![pushfreeze](https://cloud.githubusercontent.com/assets/5390448/12949981/e6befb72-d01b-11e5-9af9-70740a7353e3.png)
